### ISSUE TITLE
Property 'name' does not exist on type 'Partial<Pick<AccountInfo, "homeAccountId" | "localAccountId" | "username">>'.ts(2339)

### ISSUE THREAD

**Issue by BretMishler** [AzureAD/microsoft-authentication-library-for-js#3748]
### Core Library

MSAL.js v2 (@azure/msal-browser)

### Core Library Version

2.14.2

### Wrapper Library

MSAL React (@azure/msal-react)

### Wrapper Library Version

1.0.0

### Description

In the following code

```
import { useMsal } from "@azure/msal-react";

function ProfileContent() {
    const { instance, accounts } = useMsal();

    const name = accounts[0] && accounts[0].name;
```

JavaScript has no problem with this, and the code works as expected. But when I make this a TypeScript file, I get errors that

Property 'name' does not exist on type 'Partial<Pick<AccountInfo, "homeAccountId" | "localAccountId" | "username">>

### MSAL Configuration

```javascript
/*
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License.
 */

import { LogLevel } from "@azure/msal-browser";

/**
 * Configuration object to be passed to MSAL instance on creation. 
 * For a full list of MSAL.js configuration parameters, visit:
 * https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-browser/docs/configuration.md 
 */
export const msalConfig = {
    auth: {
        clientId: azClient,
        authority: azAuthority,
        redirectUri: "http://localhost:3000/"
    },
    cache: {
        cacheLocation: "sessionStorage", // This configures where your cache will be stored
        storeAuthStateInCookie: false, // Set this to "true" if you are having issues on IE11 or Edge
    },
    system: {	
        loggerOptions: {	
            loggerCallback: (level, message, containsPii) => {	
                if (containsPii) {		
                    return;		
                }		
                switch (level) {		
                    case LogLevel.Error:		
                        console.error(message);		
                        return;		
                    case LogLevel.Info:		
                        console.info(message);		
                        return;		
                    case LogLevel.Verbose:		
                        console.debug(message);		
                        return;		
                    case LogLevel.Warning:		
                        console.warn(message);		
                        return;
                    default:
                }	
            }	
        }	
    }
};

/**
 * Scopes you add here will be prompted for user consent during sign-in.
 * By default, MSAL.js will add OIDC scopes (openid, profile, email) to any login request.
 * For more information about OIDC scopes, visit: 
 * https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-permissions-and-consent#openid-connect-scopes
 */
export const loginRequest = {
    scopes: ["User.Read"]
};

/**
 * Add here the scopes to request when obtaining an access token for MS Graph API. For more information, see:
 * https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-browser/docs/resources-and-scopes.md
 */
export const graphConfig = {
    graphMeEndpoint: "https://graph.microsoft.com/v1.0/me"
};
```

### Relevant Code Snippets

```javascript
import { useMsal } from "@azure/msal-react";

function ProfileContent() {
    const { instance, accounts } = useMsal();

    const name = accounts[0] && accounts[0].name;
```

### Reproduction Steps

Try to execute the code snippet in ts/tsx.

### Expected Behavior

That the name property would appear.

### Identity Provider

Azure AD / MSA

### Browsers Affected (Select all that apply)

Chrome, Firefox, Edge, Safari, Internet Explorer

### Source

External (Customer)

**Comment by BretMishler** [comment]
A StackOverflow user has contributed the following:

> Browsing through the source code, it looks like they might have used the wrong type in 
> IMsalContext. There's a commit on the dev branch with the message "Fix accounts type".
> 
> Here is the current released version, the dev branch version, the pull request, and the associated issue.


https://stackoverflow.com/a/67964536/5137782

**Comment by BretMishler** [comment]
Also of note is that the code snippet shared in this post is that provided by Microsoft docs

https://docs.microsoft.com/en-us/azure/active-directory/develop/tutorial-v2-react

So I imagine a lot of people will run into this problem when they go to update their App.js to .ts.

**Comment by samuelkubai** [comment]
@BretMishler we are aware of this and have the fix for it in pull request (#3677), the fix will be available in the next release. Till then you can use the `useAccount` hook to get the account.
